utils/dom.js

```diff
@@ -10,16 +10,30 @@ import tinymce from 'tinymce';
 const { getComputedStyle, DOMRect } = window;
 const { TEXT_NODE, ELEMENT_NODE } = window.Node;
 
+/**
+ * Returns true if there is a selection and if the selection is collapsed, or
+ * false otherwise.
+ *
+ * @return {boolean} Whether there is a collapsed selection.
+ */
+export function hasCollapsedSelection() {
+	const selection = window.getSelection();
+
+	return (
+		selection.rangeCount > 0 &&
+		selection.getRangeAt( 0 ).collapsed
+	);
+}
+
 /**
  * Check whether the caret is horizontally at the edge of the container.
```

Comment:
Might want to update "caret" to "selection", as "caret" implies a collapsed selection I think.